registry/session.go

```diff
@@ -222,10 +222,10 @@ func (r *Session) GetRemoteTags(registries []string, repository string, token []
 		logrus.Debugf("Got status code %d from %s", res.StatusCode, endpoint)
 		defer res.Body.Close()
 
-		if res.StatusCode != 200 && res.StatusCode != 404 {
-			continue
-		} else if res.StatusCode == 404 {
+		if res.StatusCode == 404 {
 			return nil, fmt.Errorf("Repository not found")
+		} else if res.StatusCode != 200 {
```

Comment:
@thaJeztah could be but I'm not really sure to separate this `else if` because then you have to keep attention not move the second if statement while refactoring. Better keep them coupled but just check once